File: core/src/main/java/com/ustadmobile/core/view/SelectMultipleProductTypeTreeDialogView.java
package com.ustadmobile.core.view;


import com.ustadmobile.lib.db.entities.SaleProduct;

import java.util.List;

/**
 * SelectMultipleTreeDialog Core View extends Core UstadView. Will be implemented
 * on various implementations.
 */
public interface SelectMultipleProductTypeTreeDialogView extends UstadView {

    String VIEW_NAME = "SelectMultipleProductTypeTreeDialog";

    String ARG_PRODUCT_SELECTED_SET = "ProductSelected";

    void populateTopProductType(List<SaleProduct> locations);

    void setTitle(String title);

    /**
     * For Android: closes the activity.
     */
    void finish();

}

File: core/src/commonMain/kotlin/com/ustadmobile/core/view/SelectMultipleProductTypeTreeDialogView.kt
package com.ustadmobile.core.view


import com.ustadmobile.lib.db.entities.SaleProduct

/**
 * SelectMultipleTreeDialog Core View extends Core UstadView. Will be implemented
 * on various implementations.
 */
interface SelectMultipleProductTypeTreeDialogView : UstadView {

    fun populateTopProductType(productTypes: List<SaleProduct>)

    fun setTitle(title: String)

    /**
     * For Android: closes the activity.
     */
    fun finish()

    companion object {

        val VIEW_NAME = "SelectMultipleProductTypeTreeDialog"

        val ARG_PRODUCT_SELECTED_SET = "ProductSelected"
    }

}
